### PR TITLE
PR: Change way to check for updates

### LINKED ISSUE #7452: Check for new conda packages in Anaconda's repos instead of Github
Right now we're looking for new releases only on Github. But this gives a bad UX in Anaconda because they take several days to create new conda packages for Spyder.

So in that case we need to look for new packages (for example) in

https://repo.continuum.io/pkgs/main/win-64/

to only show the update message when new packages are placed there.

@dalthviz, please work on this one.

### PR DESCRIPTION
**Pull request by dalthviz**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP8](https://www.python.org/dev/peps/pep-0008/) for code style
* [x] Ensured your pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] Wrote at least one-line docstrings for any new functions
* [x] Added at least one unit test covering the changes, if at all possible
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed
* [x] Included a screenshot, if this PR makes any visible changes to the UI

## Description of Changes

Improve update notifications by checking agains anaconda instead of github.

### Issue(s) Resolved

Fixes #7452

### CODE REVIEW

File: spyder/workers/updates.py [hunk 1]
```diff
@@ -59,7 +60,7 @@ def check_update_available(self, version, releases):
 
     def start(self):
         """Main method of the WorkerUpdates worker"""
-        self.url = 'https://api.github.com/repos/spyder-ide/spyder/releases'
+        self.url = 'https://repo.continuum.io/pkgs/main/win-64/repodata.json'
```
Comment (ccordoba12) [inline spyder/workers/updates.py]: This looks good, thanks @dalthviz! Two comments:

1. We need to use this only for Anaconda, and use the old method for pip.
2. The repodata.json needs to be pulled from the channel corresponding the operating system in which Spyder is running.

File: spyder/workers/updates.py [hunk 2]
```diff
@@ -60,7 +66,16 @@ def check_update_available(self, version, releases):
 
     def start(self):
         """Main method of the WorkerUpdates worker"""
-        self.url = 'https://repo.continuum.io/pkgs/main/win-64/repodata.json'
+        if is_anaconda():
+            self.url = 'https://repo.continuum.io/pkgs/main'
```
Comment (goanpeca) [inline spyder/workers/updates.py]: This should be now 
`https://repo.anaconda.com/pkgs/main`

File: spyder/workers/updates.py [hunk 3]
```diff
@@ -82,14 +97,19 @@ def start(self):
...
+
+                if is_anaconda():
+                    releases = []
+                    for item in data['packages']:
+                        if 'spyder' in item and 'spyder-kernels' not in item:
+                            releases.append(item.split('-')[1])
+                    result = self.check_update_available(version, releases)
+                else:
+                    releases = [item['tag_name'].replace('v', '')
+                                                            for item in data]
```
Comment (ccordoba12) [inline spyder/workers/updates.py]: Please align this like this

```python
releases = [item['tag_name'].replace('v', '')
            for item in data]
```

**Comment by ccordoba12** [comment]
@dalthviz, please add a test for this.

**Comment by goanpeca** [comment]
(@ccordoba12 we need to change circleci to version 2.0 within the month)

**Comment by ccordoba12** [comment]
I know, I'll start working on it during the weekend.

**Comment by ccordoba12** [comment]
@dalthviz, please merge with 3.x to fix the errors in Appveyor.

File: spyder/workers/tests/test_update.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    """Test the update checking for a version that needs an update."""
+    worker = WorkerUpdates(None, False, version="1.0.0")
+    worker.start()
+    assert worker.update_available
+    assert check_version("1.0.0", worker.latest_release, '<')
+
+
+def test_not_update(qtbot):
+    """Test the update checking for a version that don't needs an update."""
+    worker = WorkerUpdates(None, False, version="10.0.0")
```
Comment (ccordoba12) [inline spyder/workers/tests/test_update.py]: Let's increase this to a very big version, e.g. `1000.0.0`

**Pull request merged**